src/psyclone/psyir/nodes/intrinsic_call.py

```diff
@@ -808,7 +1320,8 @@ class Intrinsic(IAttr, Enum):
                 types=DataNode,
                 arg_names=(("x",),)),
             optional_args={"y": DataNode, "kind": DataNode},
-            return_type=None,
+            # TODO #1590 Complex numbers unsupported.
+            return_type=lambda node: UnresolvedType(),
```

Comment:
I think should be `UnsupportedType` (and throughout?)

Reply:
I've changed these to be `UnsupportedFortranType("")` where appropriate, as the other is an abstract class. We have no original declaration here, but since this is only used as a return type I think we'll be ok.

Reply:
I do have minor concerns that using `UnsupportedFortranType` does mean it would be possible to incorrectly assume that this is a useable type (with a string of a type), however I think that should be left up to transformations using this functionality to worry about?